src/network/http_server.rs

```diff
@@ -44,6 +44,27 @@ mod serdebase64 {
     }
 }
 
+mod serdehex {
+    use hex;
+    use serde::Deserialize;
+    use serde::{de::Error, Deserializer, Serializer};
+    /// Serialize a byte vector to a "0x"‑prefixed hex string.
+    pub fn serialize<S: Serializer>(v: &Vec<u8>, s: S) -> Result<S::Ok, S::Error> {
+        // hex::encode turns &[u8] → lowercase hex (no prefix)
+        let mut prefixed = String::with_capacity(v.len() * 2 + 2);
+        prefixed.push_str("0x");
+        prefixed.push_str(&hex::encode(v));
+        s.serialize_str(&prefixed)
+    }
+    /// Deserialize from either a "0x"‑prefixed or raw hex string into Vec<u8>.
```

Comment:
Why is 0x optional? Is this behavior taken from hubs?

Reply:
Hubs seemed to be returning them _without_ the prefix so thought it was best to handle either